src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/UpsertDocument.cs

```diff
@@ -35,135 +35,149 @@ NpgsqlTransaction transaction
                         DocumentId: newDocumentId,
-                        ReferentialId: upsertRequest.DocumentInfo.SuperclassReferentialId.Value.Value,
-                        ReferentialPartitionKey: PartitionKeyFor(upsertRequest.DocumentInfo.SuperclassReferentialId.Value).Value
+                        ReferentialId: superclassIdentity.ReferentialId.Value,
+                        ReferentialPartitionKey: PartitionKeyFor(superclassIdentity.ReferentialId).Value
                     ),
                     connection,
                     transaction
                 );
             }
         }
-        catch (PostgresException pe)
+        catch (PostgresException pe) when (pe.SqlState == PostgresErrorCodes.UniqueViolation)
         {
-            if (pe.SqlState == PostgresErrorCodes.SerializationFailure)
-            {
-                _logger.LogDebug(
-                    pe,
-                    "Transaction conflict on Aliases table insert - {TraceId}",
-                    upsertRequest.TraceId
-                );
-                return new UpsertResult.UpsertFailureWriteConflict();
-            }
-
-            if (pe.SqlState == PostgresErrorCodes.UniqueViolation)
-            {
-                _logger.LogInformation(
-                    "Failure: alias identity already exists - {TraceId}",
-                    upsertRequest.TraceId
-                );
+            _logger.LogInformation(
+                pe,
+                "Failure: alias identity already exists - {TraceId}",
+                upsertRequest.TraceId
+            );
 
-                return new UpsertResult.UpsertFailureIdentityConflict(
-                    upsertRequest.ResourceInfo.ResourceName.Value,
-                    upsertRequest.DocumentInfo.DocumentIdentity.DocumentIdentityElements.Select(d =>
-                        new KeyValuePair<string, string>(d.IdentityJsonPath.Value.Substring(d.IdentityJsonPath.Value.LastIndexOf('.') + 1), d.IdentityValue)
+            return new UpsertResult.UpsertFailureIdentityConflict(
+                upsertRequest.ResourceInfo.ResourceName.Value,
+                upsertRequest.DocumentInfo.DocumentIdentity.DocumentIdentityElements.Select(
+                    d => new KeyValuePair<string, string>(
+                        d.IdentityJsonPath.Value.Substring(d.IdentityJsonPath.Value.LastIndexOf('.') + 1),
+                        d.IdentityValue
                     )
-                );
-            }
+                )
+            );
+        }
+
+        DocumentReference[] documentReferences = upsertRequest.DocumentInfo.DocumentReferences;
 
-            _logger.LogError(pe, "Failure on Aliases table insert - {TraceId}", upsertRequest.TraceId);
-            return new UpsertResult.UnknownFailure("Upsert failure");
+        if (documentReferences.Length > 0)
+        {
+            // Next insert into References
+            int numberOfRowsInserted = await _sqlAction.InsertReferences(
```

Comment:
Reference insert on upsert-as-insert.  This is where reference validation happens. An FK constraint violation means a validation failure.